### PR TITLE
Add comprehensive unit tests for image processing functions

### PR DESCRIPTION
**Pull request by xingzihai**
## Summary

This PR adds extensive test coverage for `keras.ops.image` functions, focusing on edge cases, invalid inputs, round-trip conversions, and data type handling that were not covered by existing tests.

## Changes

Added 14 new test classes with 894 lines of test code:

### New Test Classes

1. **RGBToGrayscaleEdgeCasesTest** (10 tests)
   - Single pixel conversion
   - All black/white images
   - Pure RGB colors
   - Relative brightness verification (G > R > B)
   - uint8 dtype support
   - Batch consistency

2. **RGBHSVRoundTripTest** (9 tests)
   - RGB → HSV → RGB round-trip
   - HSV → RGB → HSV round-trip
   - Pure colors (red, green, blue)
   - Black/white colors
   - Grayscale colors (saturation=0 verification)
   - Batch consistency

3. **ResizeEdgeCasesTest** (10 tests)
   - Single pixel resize
   - Resize to single pixel
   - Identity resize
   - Large up/downsampling
   - Non-square resize
   - Value range preservation
   - Aspect ratio options
   - Invalid size error handling

4. **AffineTransformEdgeCasesTest** (5 tests)
   - Identity transform
   - Translation transform
   - Small images
   - Single pixel
   - Batched transforms

5. **PadImagesEdgeCasesTest** (8 tests)
   - Zero padding
   - Asymmetric padding
   - Single pixel
   - Large padding
   - Target dimensions
   - Error handling for negative padding

6. **CropImagesEdgeCasesTest** (7 tests)
   - Zero cropping
   - Asymmetric cropping
   - Target dimensions
   - Minimum crop (to 1x1)
   - Error handling for invalid inputs

7. **GaussianBlurEdgeCasesTest** (6 tests)
   - Small/large sigma
   - Asymmetric kernel
   - Single pixel
   - Mean preservation
   - Uniform image handling

8. **ElasticTransformEdgeCasesTest** (5 tests)
   - Zero/small alpha
   - Reproducibility with seed
   - Different seeds produce different results
   - Small images

9. **PerspectiveTransformEdgeCasesTest** (3 tests)
   - Identity transform
   - Small perspective shift
   - Single pixel

10. **ExtractPatchesEdgeCasesTest** (4 tests)
    - Small images
    - Full image coverage
    - Overlapping patches
    - Single pixel

11. **MapCoordinatesEdgeCasesTest** (3 tests)
    - Identity mapping
    - Single pixel
    - Coordinate shift

12. **ScaleAndTranslateEdgeCasesTest** (4 tests)
    - Identity operation
    - Small/large scale
    - Translation

13. **ImageOpsInvalidInputTest** (7 tests)
    - Wrong dtypes (int for HSV functions)
    - Wrong size types
    - Conflicting arguments
    - Coordinate rank mismatch

14. **ImageOpsChannelsFirstEdgeCasesTest** (7 tests)
    - Channels_first format support for all major functions

## Testing

All tests follow the existing test patterns in the repository:
- Use `testing.TestCase` base class
- Proper setUp/tearDown for data format
- Use parameterized tests where appropriate
- Clear test names and docstrings

## Checklist

- [x] Tests follow existing patterns in `image_test.py`
- [x] All test classes have proper setUp/tearDown
- [x] Tests cover edge cases, invalid inputs, and different data formats
- [x] Clear commit message describing changes

### CODE REVIEW

File: keras/src/ops/image_test.py [hunk 1]
```diff
@@ -2893,3 +2893,897 @@ def test_extract_patches_3d_non_cubic(self, dtype, data_format):
             volume, size=(2, 3, 4), strides=(2, 3, 4), data_format=data_format
         )
         self.assertEqual(patches.shape, expected_shape)
+
+
+class RGBToGrayscaleEdgeCasesTest(testing.TestCase):
```
Comment (hertschuh) [inline keras/src/ops/image_test.py:2898]: These tests significantly overlap the existing ones. Most of them can be removed.

Additionally, these tests should be interspersed with the existing ones so that all tests for the same method are grouped together.

**Review by hertschuh — CHANGES_REQUESTED** [review]
Please also approve the CLA
